Dockerfile

```diff
@@ -1,46 +1,23 @@
-FROM rust:1.71.1 as builder
+FROM lukemathwalker/cargo-chef:latest-rust-1.73 AS chef
+WORKDIR /app
 
-# create a new empty shell project
-RUN USER=root cargo new --bin hm-discord-bot
-WORKDIR /hm-discord-bot
+FROM chef AS planner
+COPY . .
+RUN cargo chef prepare --recipe-path recipe.json
 
-# Copy manifests
-COPY ./Cargo.lock ./Cargo.lock
-COPY ./Cargo.toml ./Cargo.toml
+FROM chef AS builder
+COPY --from=planner /app/recipe.json recipe.json
+# Build dependencies - this is the caching Docker layer!
+RUN cargo chef cook --release --recipe-path recipe.json
+# Build application
+COPY . .
+RUN SQLX_OFFLINE=true cargo build --release
 
-# Build only the dependencies to cache them
-RUN cargo build --release
-RUN rm src/*.rs
+FROM debian:bookworm-slim AS runtime
+RUN apt update && apt install ca-certificates openssl -y && rm -rf /var/lib/apt/lists/*
 
-# Now that the dependency is built, copy your source code
-COPY ./migrations ./migrations
-COPY ./src ./src
+COPY --from=builder /app/target/release/hm-discord-bot /usr/local/bin/hm-discord-bot
 
-# Build for release.
-RUN rm ./target/release/deps/hm_discord_bot*
-RUN cargo install --path .
+WORKDIR /usr/local/hm-discord-bot
```

Comment:
change workdir back to `/hm-discord-bot` since a short workdir makes volume mapping easier to understand, also there is less of a risk to mix up host and container.